File: cleanup/conditional/conditional.py
from secrets import randbits

class Mess:
    def __init__(self, size):
        print(f"Make a {size} mess.")
        self.size = size

    def cleanup(self):
        print(f"Clean up the {self.size} mess.")

    def look(self):
        print(f"Look at the {self.size} mess!")

    def __enter__(self):
        return self

    def __exit__(self, *_):
        self.cleanup()

def random_mess(mess1, mess2):
    if randbits(1):
        mess2.cleanup()
        return mess1
    else:
        mess1.cleanup()
        return mess2

mess1 = Mess("first")
mess2 = Mess("second")
with random_mess(mess1, mess2) as mess:
    mess.look()

File: cleanup/conditional/conditional.rs
struct Mess {
    size: String,
}

impl Mess {
    fn new(size: String) -> Mess {
        println!("Make a {} mess.", size);
        Mess { size }
    }

    fn look(&self) {
        println!("Look at the {} mess!", self.size);
    }
}

impl Drop for Mess {
    fn drop(&mut self) {
        println!("Clean up the {} mess.", self.size);
    }
}

fn random_mess(mess1: Mess, mess2: Mess) -> Mess {
    if rand::random() {
        mess1
    } else {
        mess2
    }
}

fn main() {
    let mess1 = Mess::new("first".into());
    let mess2 = Mess::new("second".into());
    let mess = random_mess();
    mess.look()
}
